### PR TITLE
Convert hyphen (-) to dot (.) in version string

### PR DESCRIPTION
**Pull request by jacobq**
This PR is intended to address https://github.com/unindented/electron-installer-redhat/issues/36
After reading the version number option but before performing creating the package, a few additional lines of code check whether or not the version string contains a hyphen (-) character. If it does then a warning is generated and the hyphen is replaced with a dot (.) character. This is because otherwise an error will occur when attempting to run `rpmbuild`, e.g.
```
Error executing command (1): 
rpmbuild -bb /tmp/electron-11729-4640-1kh8cdn.sxtbne9udi/ember-electron_2.0.0-beta_x86_64/SPECS/ember-electron.spec --target x86_64
error: line 2: Illegal char '-' in: Version: 2.0.0-beta
```

I could not find a definite reference for the exact syntax allowed for this field of the SPEC file, but it seems that hyphens are definitely not allowed yet they are common/standard in [semantic versioning](http://semver.org/), so they should not cause the build to fail.

### CODE REVIEW

**Comment by unindented** [comment]
This looks good to me. Have you tried it on your package?

**Comment by jacobq** [comment]
No, in my app I set the parameter explicitly, but to test I did write a test that fails without the change and passes with the change.

**Comment by unindented** [comment]
Yup, I was wondering about a real-world scenario. But the test looks good, so I'll merge.

**Comment by jacobq** [comment]
Understood. The functionality should be identical, but I will test later to verify in another app.

File: src/installer.js [hunk 1]
```diff
@@ -384,6 +384,12 @@ module.exports = function (data, callback) {
     async.apply(getDefaults, data),
     async.apply(getOptions, data),
     function (options, callback) {
+      let adjustedVersion = options.version.replace(/-/, '.')
```
Comment (unindented) [inline src/installer.js]: This could be a `const`.

**Comment by unindented** [comment]
What would you think about adopting a similar solution to what was proposed in `electron-installer-debian`: https://github.com/unindented/electron-installer-debian/blob/45a6618a688aaf9d8674f05a5d59328903712c68/src/installer.js#L121

**Comment by jacobq** [comment]
If I knew the precise syntax/format permitted for RPM SPEC files I would want to do something like that, but without one I'd just be guessing. Though even so, once regex starts to get that long I don't think it's good to have inline any more since it really ought to have its own tests. Plus in my case the version string might be anything that is semver 2.0 compliant, so there might be more after the hyphen than that short list (rc, alpha, beta, etc. -- could be snapshot, fork, build number, etc.).

**Comment by unindented** [comment]
Fair enough.

**Pull request merged**